### ISSUE TITLE
Validate relation after changing road segments

### ISSUE THREAD

**Issue by tordans** [openstreetmap/iD#5661]
Recently I added the roudabout at https://www.openstreetmap.org/changeset/65847412 after the street was remodeled but forgot to add the Bus-Route-Relation and Street-Network-Relation to the new road segment (roundabout).

Luckily someone spottet it with [tools.geofabrik.de-pubtrans_routes](http://tools.geofabrik.de/osmi/?view=pubtrans_routes&lon=6.37433&lat=50.81467&zoom=15&overlays=ptv2_routes_,ptv2_routes_valid,ptv2_routes_invalid,ptv2_error_,ptv2_error_ways,ptv2_error_nodes) and fixed it in the changeset above.

# Suggestion

Add a validation warning similar to the "there are unconnected roads" before I commit changes.

The warning should check: Does the area include relations. Did I download a closed relation. And is this closed relation now open/broken. If so, show a warning like "You broke the relation, please close it again or skip this warning."

Apparently JOSM has such a behaviour.

**Comment by slhh** [comment]
@tordans 
A typical iD user would unlikely be able to fix the broken relation, and would likely have to discard the full changeset. This isn't nice.

We need to flag the issue as soon as it occurs, and we need to prevent breaking the route stucture at all.
This doen't mean to prevent the breaking operations, but to fix the route structure automatically.
We can do this by adding a temporary way closing a generated gap of a route. The way wouldn't be tagged as a highway, but with a special tag to indicate it's a gap segment. The gap segment should be rendered using an eye-catching error color to flag the issue.

Preferably, the gap segments should be dragable onto highways (or railway etc. dependent on route type). When they become coincident, the gap segment should be automatically deleted and route membership should be transfered to the highway.

Adding gap segments does potentially geneate coincidence of ways, therefore, we need to fix #1239 as a prerequisite.

**Comment by tordans** [comment]
> A typical iD user would unlikely be able to fix the broken relation, and would likely have to discard the full changeset. This isn't nice.

Yes, it not ideal. However, currently people like me break and then upload broken stuff. That is even less ideal. For the map, but also for me, 'cause I feel bad for breaking stuff. Personally, I would rather not commit than break stuff, given the choice.

> … but to fix the route structure automatically.
We can do this by adding a temporary way closing a generated gap of a route.

I was hoping for an approach that requires far less work as an MVP. What you describe will be perfect, once it fully works, but it is a huge effort with a lot of open questions, eg. the required "special tagging".

--- 

@bhousel @quincylvania now that live validations are in place, do you see a solution to this based on the live validation?

I don't know how we could get numbers for this issue (like "how often do iD users break relations like I did and described"). So its hard to estimate the priority/urgency of at least a warning-solution.

**Comment by bhousel** [comment]
> @bhousel @quincylvania now that live validations are in place, do you see a solution to this based on the live validation?

Yes we should validate relations as users edit.
(@slhh is not on the development team and does not speak for us)
